src/textual/geometry.py

```diff
@@ -470,11 +484,100 @@ def union(self, region: Region) -> Region:
         x1, y1, x2, y2 = self.corners
         ox1, oy1, ox2, oy2 = region.corners
 
-        union_region = Region.from_corners(
+        union_region = self.from_corners(
             min(x1, ox1), min(y1, oy1), max(x2, ox2), max(y2, oy2)
         )
         return union_region
 
+    def split(self, cut_x: int, cut_y: int) -> tuple[Region, Region, Region, Region]:
+        """Split a region in to 4 from given x and y offsets (cuts).
+
+                   cut_x ↓
```

Comment:
Love these diagrams, they help so much.

If docs are generated from these docstrings we'll need to check how to have this diagram be printed inside `<pre>` tags or something